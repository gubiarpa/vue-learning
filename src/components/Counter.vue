<template>
    <h2>Counter</h2>
    <div>
        {{ counter }} <sup>2</sup> = {{ squareValue }}
    </div>
    <div class="buttons-container">
        <button @click="increase">Increase</button>
        <button @click="decrease">Decrease</button>
    </div>
</template>

<script>
export default {
    // name: 'Wawitash',
    data() {
        return {
            counter: 5
        }
    },
    computed: {
        squareValue() {
            return this.counter ** 2
        }
    },
    methods: {
        getSquareValue() {
            return this.counter ** 2
        },
        increase() {
            this.counter++
        },
        decrease() {
            this.counter--
        },
    },
}
</script>

<style>

</style>