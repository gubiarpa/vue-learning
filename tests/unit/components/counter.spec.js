import { shallowMount, mount } from '@vue/test-utils'
import Counter from '@/components/Counter'

describe('Counter Component', () => {
    test('Debe hacer el match con el snapshot', () => {
        const wrapper = shallowMount(Counter)

        expect(wrapper.html()).toMatchSnapshot()
    })
})