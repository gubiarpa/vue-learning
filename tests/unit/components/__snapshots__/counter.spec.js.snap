// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe hacer el match con el snapshot 1`] = `
<h2>Counter</h2>
<div>5 <sup>2</sup> = 25</div>
<div class="buttons-container"><button>Increase</button><button>Decrease</button></div>
`;
